Add instructions on using a Docker image for development

joshessman-llnl opened the pull request:
Resolves #328

https://serac.readthedocs.io/en/docs-essman-using_docker/sphinx/dev_guide/docker_dev_env.html

joshessman-llnl committed:
docs: added instructions on using a Docker image for development

jamiebramwell commented:
The docs are available for browsing [here](https://serac.readthedocs.io/en/docs-essman-using_docker/).

white238 commented on src/docs/sphinx/docker_info.rst:
```diff
@@ -7,6 +7,37 @@
+.. image:: pull_complete.png
+
+5. You can now run the image.  Run ``docker run -it -v /your/serac/repo:/home/serac/serac seracllnl/tpls:gcc-8_12-15-20_20h-36m bash``,
+   replacing the tag (the part following the ``:``) with the tag you used in the ``docker pull`` command and
```
This instruction is confusing, mainly due to two `:` in the command.

white238 commented on src/docs/sphinx/docker_info.rst:
```diff
@@ -7,6 +7,37 @@
+#. Follow the build instructions detailed in :ref:`building-serac`, using the host-config in ``host-configs/docker`` that
+   corresponds to the compiler you've selected.  These commands should be run using the terminal you opened in the previous step.
+
+#. You can now develop normally your host machine (e.g., via a graphical text editor).
```
"normally **on** your host machine"?

Also somewhere this needs clarification that you are mapping the local directory to the directory in the docker container

joshessman-llnl committed:
docs: clarification on development workflow, note on the use of docker volumes

jamiebramwell commented:
When I tried to do this, I got a permissions error. Any ideas? 
<img width="866" alt="Screenshot 2021-01-15 135033" src="https://user-images.githubusercontent.com/17935880/104782219-b4eabd80-5738-11eb-8f09-4b47aa47a19c.png">

joshessman-llnl commented:
> 
> 
> When I tried to do this, I got a permissions error. Any ideas?
> <img alt="Screenshot 2021-01-15 135033" width="866" src="https://user-images.githubusercontent.com/17935880/104782219-b4eabd80-5738-11eb-8f09-4b47aa47a19c.png">

Thanks for trying this out!  I think there may need to be a `-u serac` in the `docker run` command - I'll update the docs accordingly.

joshessman-llnl committed:
docs: update docker docs to specify user when starting container

jamiebramwell committed:
doc: change the docker instructions to use the axom user

jamiebramwell reviewed (approved):
Works on my machine! We should figure out why the volume is mounted as axom instead of serac in the docker container for a future PR.

jamiebramwell commented:
@white238 Do you have any outstanding issues?

joshessman-llnl committed:
Merge branch 'develop' into docs/essman/using_docker

joshessman-llnl committed:
Merge branch 'develop' into docs/essman/using_docker

joshessman-llnl committed:
fix: move new documentation to developer guide

jamiebramwell commented:
@jandrej @salazardetroya , since you are interested in using these instructions, does it make sense? Thanks for the help!

joshessman-llnl committed:
fix: move docker screenshots to correct location

jamiebramwell commented:
The documentation is built and hosted here: https://serac.readthedocs.io/en/docs-essman-using_docker/sphinx/dev_guide/docker_dev_env.html

joshessman-llnl committed:
Merge branch 'develop' into docs/essman/using_docker

jamiebramwell committed:
Merge branch 'develop' into docs/essman/using_docker

jamiebramwell committed:
docs: Add more explicit directions with a note about bind mount permissions

jamiebramwell committed:
docs: Fix the spacing for the code block

jamiebramwell committed:
Merge branch 'develop' into docs/essman/using_docker

salazardetroya commented:
Hi, I've got serac and compiled and running with the docker image! I would only stress downloading **the most recent** image in the docs to prevent lazy people like me to simply copy and paste the instructions given there. Or maybe add an image with the `latest` tag? That is what I usually see in other projects.

jamiebramwell commented:
An excellent point @salazardetroya ! I'll update the docs and look into adding a `latest` tag. Thanks for trying it out!

jamiebramwell committed:
Merge branch 'develop' into docs/essman/using_docker

joshessman-llnl committed:
ci: add xsltproc to docker, tag with compiler-specific latest tag

joshessman-llnl committed:
ci: use new docker images with latest tag

joshessman-llnl committed:
ci: switch to real docker image

joshessman-llnl commented:
I went ahead and updated the docker images to add `latest` tags for each compiler version, and since I was updating the images, I added `xsltproc` as well: https://dev.azure.com/llnl-serac/serac/_build/results?buildId=2932&view=logs&j=2742912e-f547-59b4-f336-65c28301f30c&t=cd7bfce2-9b19-5c4f-5f3a-913e42776d4e&l=12418

Resolves #359

jamiebramwell commented:
@white238 can you take a look at this? I think it's been vetted enough to merge at this point.

white238 commented:
LGTM
